server/core/store.go

```diff
@@ -311,16 +312,18 @@ type StoreHotRegionInfos struct {
 // StoreHotRegionsStat used to record the hot region statistics group by store
 type StoreHotRegionsStat map[uint64]*HotRegionsStat
 
-var (
-	// ErrStoreNotFound is for log of store no found
-	ErrStoreNotFound = func(storeID uint64) error {
-		return errors.Errorf("store %v not found", storeID)
-	}
-	// ErrStoreIsBlocked is for log of store is blocked
-	ErrStoreIsBlocked = func(storeID uint64) error {
-		return errors.Errorf("store %v is blocked", storeID)
-	}
-)
+type errStoreNotFound struct {
+	storeID uint64
+}
+
+func (e errStoreNotFound) Error() string {
+	return fmt.Sprintf("store %v not found", e.storeID)
+}
+
+// NewStoreNotFound is for log of store not found
+func NewStoreNotFound(storeID uint64) errcode.ErrorCode {
```

Comment:
Will `NewStoreNotFoundError` be a better name?

Reply:
Do you want to suffix all errors with `Error` then? It makes sense if you are reading the function without knowing the types. If you know the types then hungarian-like notations can be seen as boilerplate. We can also abbreviate as `Err`.